src/api/goods.ts

```diff
@@ -5,15 +5,32 @@ const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/good
 
 export function getAll(): Promise<Good[]> {
   return fetch(API_URL)
-    .then(response => response.json());
+    .then(response => {
+      if (!response.ok) {
+        throw new Error('Cannot fetch goods');
+      }
```

Comment:
you return error in `getAll` then in `handleRed` and `handle5First` try to put this error in setState variable

here is my suggestion how to prevent this behavior:
```suggestion
    if (!response.ok) {
      console.error("Cannot fetch goods");
      return [];
    }
```